scripts/EGAP/create_EGAP_json.py

```diff
@@ -0,0 +1,253 @@
+        header_row = next(csv_reader)
+        normalized_header_row = [col_header.decode('ascii', 'ignore') for col_header in header_row]
+
+        id_index = normalized_header_row.index('ID')
+        for line in csv_reader:
+            row = [cell.decode('ascii', 'ignore').strip() for cell in line]
+            project_id = row[id_index]
+            logger.info('Adding project ID: {}'.format(project_id))
+            root_directory = os.path.join(top_dir, project_id)
+            os.mkdir(root_directory)
+            data_directory = os.path.join(root_directory, 'data')
+            os.mkdir(data_directory)
+            os.mkdir(os.path.join(data_directory, 'nonanonymous'))
+            project_dict = make_project_dict(row, root_directory, author_list, normalized_header_row)
+            make_json_file(root_directory, project_dict, 'project')
+            registration_dict = make_registration_dict(row, root_directory, normalized_header_row)
+            make_json_file(root_directory, registration_dict, 'registration')
+
+def create_author_dict():
+    # Reads in author CSV and returns a list of dicts with names and emails of EGAP Authors
+    authors = []
+    with open('EGAP_author_emails.csv') as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        header_row = next(csv_reader)
+        normalized_header_row = [col_header.decode('ascii', 'ignore') for col_header in header_row]
+
+        name_index = normalized_header_row.index('Name')
+        email_index = normalized_header_row.index('Email')
+        for line in csv_reader:
+            row = [cell.decode('ascii', 'ignore').strip() for cell in line]
+            logger.info('Adding user: ' + row[name_index])
+            if row[email_index] != '':
+                author_dict = {'name': row[name_index].strip(), 'email': row[email_index]}
+            else:
+                author_dict = {'name': row[name_index].strip()}
+            authors.append(author_dict)
+    return authors
+
+def make_project_dict(row, author_list, normalized_header_row):
+    project = {}
+    title_index = normalized_header_row.index('TITLE')
+    id_index = normalized_header_row.index('ID')
+    postdate_index = normalized_header_row.index('POST DATE')
+    contributors_index = normalized_header_row.index('B2 AUTHORS')
+    project['id'] = row[id_index]
+    project['title'] = row[title_index]
+    project['post-date'] = row[postdate_index]
+
+    authors = row[contributors_index]
+
+    authors = authors.split(',')
+    project['contributors'] = []
+    author_name_list = [author['name'] for author in author_list]
+    for author in authors:
+        author = author.strip()
+        if author:
+            matched_authors = [e for e in author_name_list if e.startswith(author)]
+            if not matched_authors:
+                logger.info('Author {} not in Author spreadsheet...'.format(author))
+                # project['contributors'].append(author)
```

Comment:
Should this comment be here?

Reply:
It should not. Must have forgotten to save!